dotCMS/src/main/java/com/dotmarketing/viewtools/navigation/NavTool.java

```diff
@@ -122,33 +122,36 @@ protected static NavResult getNav(Host host, String path, long languageId, User
                     children.add(nav);
                 }
                 else if(item instanceof IHTMLPage) {
-                	final String httpProtocol = "http://";
-                	final String httpsProtocol = "https://";
-                    IHTMLPage itemPage=(IHTMLPage)item;
-                    ident=APILocator.getIdentifierAPI().find(itemPage);
+                    IHTMLPage itemPage = (IHTMLPage)item; 
+                    if(itemPage.getLanguageId() == languageId) {
+                        final String httpProtocol = "http://";
+                        final String httpsProtocol = "https://";
+
+                        ident=APILocator.getIdentifierAPI().find(itemPage);
+
+                        String redirectUri = itemPage.getRedirect();
+                        NavResult nav=new NavResult(folder.getInode(),host.getIdentifier(),languageId);
+                        nav.setTitle(itemPage.getTitle());
+                        if(UtilMethods.isSet(redirectUri) && !redirectUri.startsWith("/")){
+                            if(redirectUri.startsWith(httpsProtocol) || redirectUri.startsWith(httpProtocol)){
+                                nav.setHref(redirectUri);	
+                            }else{
+                                if(itemPage.isHttpsRequired())
+                                    nav.setHref(httpsProtocol+redirectUri);	
+                                else	
+                                    nav.setHref(httpProtocol+redirectUri);
+                            }
 
-                    String redirectUri = itemPage.getRedirect();
-                    NavResult nav=new NavResult(folder.getInode(),host.getIdentifier(),languageId);
-                    nav.setTitle(itemPage.getTitle());
-                    if(UtilMethods.isSet(redirectUri) && !redirectUri.startsWith("/")){
-                        if(redirectUri.startsWith(httpsProtocol) || redirectUri.startsWith(httpProtocol)){
-                      	  nav.setHref(redirectUri);	
                         }else{
-                      	  	if(itemPage.isHttpsRequired())
-                      	  		nav.setHref(httpsProtocol+redirectUri);	
-                    		else	
-                    			nav.setHref(httpProtocol+redirectUri);
+                            nav.setHref(ident.getURI());
+                        }
+                        nav.setOrder(itemPage.getMenuOrder());
+                        nav.setType("htmlpage");
+                        nav.setPermissionId(itemPage.getPermissionId());
+                        nav.setShowOnMenu(itemPage.isShowOnMenu());
+                        if(!itemPage.isContent() || (itemPage.isContent() && (itemPage.getLanguageId() == languageId || LanguageWebAPI.canDefaultPageToDefaultLanguage()) )) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)